usermods/usermod_v2_RF433/usermod_v2_RF433.h

```diff
@@ -0,0 +1,178 @@
+  unsigned long lastTime = 0;
+
+  bool modEnabled = true;
+  int8_t receivePin = -1;
+
+  static const char _modName[];
+  static const char _modEnabled[];
+  static const char _receivePin[];
+
+  bool initDone = false;
+
+public:
+
+  void setup()
+  {
+    mySwitch.disableReceive();
+    if (modEnabled)
+    {
+      mySwitch.enableReceive(receivePin);
+    }
+    initDone = true;
+  }
+
+  /*
+   * connected() is called every time the WiFi is (re)connected
+   * Use it to initialize network interfaces
+   */
+  void connected()
+  {
+  }
+
+  void loop()
+  {
+    if (!modEnabled || strip.isUpdating()) 
+      return;
+
+    if (mySwitch.available())
+    {
+      unsigned long value = mySwitch.getReceivedValue();
+      mySwitch.resetAvailable();
+
+      // Discard duplicates, limit long press repeat
+      if (lastValue == value && millis() - lastTime < 800)
+        return;
+
+      lastValue = value;
+      lastTime = millis();
+
+      DEBUG_PRINT(F("RF433 Receive: "));
+      DEBUG_PRINTLN(value);
+      
+      if(!remoteJson433(value))
+        DEBUG_PRINTLN(F("RF433: unknown button"));
+    }
+  }
+
+  // Add last received button to info pane
+  void addToJsonInfo(JsonObject &root)
+  {
+    if (!initDone)
+      return; // prevent crash on boot applyPreset()
+    JsonObject user = root["u"];
+    if (user.isNull())
+      user = root.createNestedObject("u");
+
+    JsonArray switchArr = user.createNestedArray("RF433 Last Received"); // name
+    switchArr.add(lastValue); // value
+  }
+
+  void addToConfig(JsonObject &root)
+  {
+    JsonObject top = root.createNestedObject(FPSTR(_modName)); // usermodname
+    top[FPSTR(_modEnabled)] = modEnabled;
+    JsonArray pinArray = top.createNestedArray("pin");
+    pinArray.add(receivePin);
+
+    DEBUG_PRINTLN(F(" config saved."));
+  }
+
+  bool readFromConfig(JsonObject &root)
+  {
+    JsonObject top = root[FPSTR(_modName)];
+    if (top.isNull())
+    {
+      DEBUG_PRINT(FPSTR(_modName));
+      DEBUG_PRINTLN(F(": No config found. (Using defaults.)"));
+      return false;
+    }
+    getJsonValue(top[FPSTR(_modEnabled)], modEnabled);
+    getJsonValue(top["pin"][0], receivePin);
+
+    DEBUG_PRINTLN(F("config (re)loaded."));
+
+    // Redo init on update
+    if(initDone)
+      setup();
+
+    return true;
+  }
+
+  /*
+   * getId() allows you to optionally give your V2 usermod an unique ID (please define it in const.h!).
+   * This could be used in the future for the system to determine whether your usermod is installed.
+   */
+  uint16_t getId()
+  {
+    return USERMOD_ID_RF433;
+  }
+
+  // this function follows the same principle as decodeIRJson()
+  bool remoteJson433(int button)
+  {
+    char objKey[14];
+    bool parsed = false;
+
+    if (!requestJSONBufferLock(22)) return false;
+
+    sprintf_P(objKey, PSTR("\"%d\":"), button);
+
+    // attempt to read command from remote.json
+    readObjectFromFile(PSTR("/remote433.json"), objKey, pDoc);
```

Comment:
Would it make more sense to read the JSON file during your setup function rather than for every button press?

Reply:
This is done the same as for IR and wizmote. If it wasn't there if you went to /edit and changed the file changes wouldn't be taken into account until a reboot? That wouldn't be user-friendly.

Reply:
I agree, this (reading the file upon receive) is the way it is done in all other json remote code. Also there is no restriction on creating say 50 different commands: that would hog a lot of RAM permanently.